src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1481,7 +1481,20 @@ class kernel_impl : public std::enable_shared_from_this<kernel_impl>
       kcmd->opcode = (protocol == control_type::fa) ? ERT_START_FA : ERT_START_CU;
       break;
     case kernel_type::dpu :
-      kcmd->opcode = (m_module ? ERT_START_DPU : ERT_START_CU);
+      if(m_module)
+      {
+        // TBD: this is temporary until ELF has save/restore sections
+        char* env = std::getenv("XRT_PREEMPT_MODE");
+        if(std::stoi(env) == 1)
+            kcmd->opcode = ERT_START_DPU_PREEMPT;
```

Comment:
Not sure we need this.

Reply:
This is to switch between ERT_START_DPU_PREEMPT vs ERT_START_DPU flow. I am getting the save/restore control code from a file. Please refer the PR in xrt-mcdm repo for that changes.